tests/integ/test_packaging.py

```diff
@@ -0,0 +1,563 @@
+        file.write(requirements)
+        requirements_path = file.name
+
+    yield requirements_path
+
+    # Clean up the temporary files after the test completes
+    for path in {requirements_path, local_script_path, new_path}:
+        if os.path.exists(path):
+            os.remove(path)
+
+
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="matplotlib required",
+)
+def test_add_requirements_with_local_filepath(
+    session, requirements_file_with_local_path
+):
+    """
+    Assert that is a requirement file references local python scripts, the variables in those local python scripts
+    are available for use within a UDF.
+    """
+    session.add_requirements(requirements_file_with_local_path)
+    assert session.get_packages() == {
+        "matplotlib": "matplotlib",
+        "pyyaml": "pyyaml==6.0",
+    }
+
+    udf_name = Utils.random_name_for_temp_object(TempObjectType.FUNCTION)
+
+    @udf(name=udf_name)
+    def use_local_file_variables() -> str:
+        from nicename import VARIABLE_IN_LOCAL_FILE
+
+        return f"{VARIABLE_IN_LOCAL_FILE + 10}"
+
+    Utils.check_answer(session.sql(f"select {udf_name}()"), [Row("60")])
+
+
+def test_add_requirements_yaml(session, resources_path):
+    test_files = TestFiles(resources_path)
+
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        session.add_requirements(test_files.test_conda_environment_file)
+    assert session.get_packages() == {
+        "numpy": "numpy==1.24.3",
+        "pandas": "pandas",
+        "scikit-learn": "scikit-learn==0.24.2",
+        "matplotlib": "matplotlib==3.7.1",
+        "seaborn": "seaborn==0.11.1",
+        "scipy": "scipy==1.10.1",
+    }
+    assert session._runtime_version_from_requirement == "3.8"
+
+    udf_name = Utils.random_name_for_temp_object(TempObjectType.FUNCTION)
+    system_version = f"{sys.version_info[0]}.{sys.version_info[1]}"
+
+    if system_version != session._runtime_version_from_requirement:
+        with pytest.raises(ValueError) as ex_info:
+
+            @udf(name=udf_name)
+            def get_numpy_pandas_version() -> str:
+                import scipy
+                import seaborn as sns
+                import tensorflow as tf
+
+                return f"{tf.__version__}/{sns.__version__}/{scipy.__version__}"
+
+        assert (
+            "Cloudpickle can only be used to send objects between the exact same version of Python. "
+            in str(ex_info)
+        )
+    else:
+
+        @udf(name=udf_name)
+        def get_numpy_pandas_version() -> str:
+            import scipy
+            import seaborn as sns
+
+            return f"{sns.__version__}/{scipy.__version__}"
+
+        Utils.check_answer(session.sql(f"select {udf_name}()"), [Row("0.11.1/1.10.1")])
+
+
+def test_add_requirements_with_bad_yaml(session, bad_yaml_file):
+    with pytest.raises(ValueError) as ex_info:
+        session.add_requirements(bad_yaml_file)
+    assert (
```

Comment:
I'm not against this but for new tests, I prefer using `match` in `pytest.raises(Exception, match='regex')`

Reply:
Updated